### DISCUSSION TITLE
Automated tests using GitHub Actions, Cypress, and Supabase CLI

### DISCUSSION THREAD

**Discussion by churichard** [supabase/supabase#10202] [Questions]
I'm trying to set up an automated CLI for running Cypress tests by leveraging the Supabase CLI. Ideally, I would want GitHub Actions to set up a new local Supabase database and for Cypress to use the database when running the tests.

However, I'm unsure how to get Cypress to use the correct Supabase URL and keys. In my real code, I populate them using environment variables, but I haven't found a way to set environment variables with the Supabase CLI.

Is there any guidance or documentation on how to do this?

**Comment by santosant** [comment]
I'm not sure but supabase CLI authentication only works through a token generated from https://app.subase.com/account/tokens it doesn't have access to create db by default like standard libraries, the cli uses the token and reference from URL of your project to communicate;

Locally you can create migrations/databases without needing environment variables, so if everything is ok you can update your production environment using the cli.

https://supabase.com/docs/reference/cli/usage#supabase-db

**Comment by jemmyw** [comment]
You can get the environment variables by running `supabase status -o env` which gives you an appropriate output to export to the shell or pass to other processes. You can also do `supabase status -o json` to get the variables in a json format.
